superx/routes/home.py

```diff
@@ -76,27 +80,49 @@ def cart():
 
 def livesearch():
     """
-    returns search functin using jquery data and ajax so not to redirect
+    returns search function using jquery data and ajax so not to redirect
     """
 
-    json_list_of_items = []
+    products = []
+    search_res = request.form.get("input").strip()
+    branches_code_list = session['branches_data']
 
     # if search_res is empty string, return empty json
-    search_res = request.form.get("input").strip()
     if not search_res:
-        return render_template('products_table.html', products=json_list_of_items)
+        return render_template('products_table.html', products=products)
```

Comment:
This would be a bit cleaner if you changed it to:
`if search_res:`
got rid of the return statement here 
and indented everything from here until the next return statement.